Code/Framework/AzGameFramework/AzGameFramework/Application/GameApplication.cpp

```diff
@@ -55,6 +55,12 @@ namespace AzGameFramework
 
         AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_TargetBuildDependencyRegistry(registry, AZ_TRAIT_OS_PLATFORM_CODENAME, specializations, &scratchBuffer);
 
+#if AZ_TRAIT_OS_IS_HOST_OS_PLATFORM && defined (AZ_DEBUG_BUILD) || defined(AZ_PROFILE_BUILD)
```

Comment:
Actually we want to group the determination of the build configuration so that it is separate from the check that this code is running on a HostOS like Windows.

```suggestion
#if AZ_TRAIT_OS_IS_HOST_OS_PLATFORM && (defined (AZ_DEBUG_BUILD) || defined(AZ_PROFILE_BUILD))
```

Reply:
Because `&&` has a higher [order of precedence](https://en.cppreference.com/w/c/language/operator_precedence) than `||` in the preprocessor, the check is currently
1. Is the the platform a Host OS(Windows, Linux, Mac) AND is the configuration Debug
2. OR is the configuration Profile

We just want the check to be
1. Is the platform a Host OS(Windows, Linux, Mac) AND is the configuration (Debug OR Profile)

Reply:
Oops. Sorry, I didn't catch it.

Reply:
No that was my fault, the snippet I provided, didn't have the parenthesis around the build configuration checks